### PR TITLE
Managing uploaded file state + maintaining clean cloned object

### PR DESCRIPTION
**Pull request by karelm**
## Description

Upload images and create a new story group card as a consequence

## Checklist

-   [ ] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] I have tested different user stories. 
-   [ ] I have identified similar or related features and tested they are still working.
-   [ ] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [ ] I have commented my code, particularly in hard-to-understand areas.

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: packages/story-group-composer/components/CarouselCards/index.jsx [hunk 1]
```diff
@@ -186,22 +185,30 @@ class CarouselCards extends React.Component {
...
                       canHaveMediaAttachmentType: () => true,
                     }}
-                    uploadType={UploadTypes.LINK_THUMBNAIL}
+                    uploadType={UploadTypes.MEDIA}
                     multiple
                     disabled={false}
                   />
                 </div>
               )}
+              {!card.empty && card.progress !== null && card.uploading && (
```
Comment (MayaUribe) [inline packages/story-group-composer/components/CarouselCards/index.jsx]: This was satisfying to test 😄

File: packages/story-group-composer/components/StoryGroupWrapper/index.jsx [hunk 2]
```diff
@@ -38,14 +38,15 @@ const StoryGroupWrapper = ({
...
   onComposerClick,
+  onCreateNewStoryCard,
+  onUpdateStoryUploadProgress,
   onSetShowDatePicker,
   showDatePicker,
   userData,
   draft,
 }) => {
-  // hooks: https://reactjs.org/docs/hooks-state.html
+  const cards = editingStoryGroup ? editingStoryGroup.storyDetails.stories : draft.stories;
```
Comment (MayaUribe) [inline packages/story-group-composer/components/StoryGroupWrapper/index.jsx]: I think we should try to consolidate this variable to be the same everywhere so it's consistent, we have `draft` and `editingStoryGroup`, they don't seem to be the same type of object although they are both referring to storyGroup.

File: packages/story-group-composer/index.js [hunk 3]
```diff
@@ -45,6 +45,24 @@ export default connect(
     onComposerClick: (showDatePicker) => {
       if (showDatePicker) dispatch(actions.setShowDatePicker(false));
     },
+    onCreateNewStoryCard: ({ id, uploaderInstance, file }) => {
```
Comment (MayaUribe) [inline packages/story-group-composer/index.js]: So we are calling this specifically for file uploading, correct? and when we click on Schedule is when we do the StoryGroup create/update with this saved state, correct? :)

File: packages/story-group-composer/reducer.js [hunk 4]
```diff
@@ -68,11 +113,43 @@ export default (state = initialState, action) => {
...
+    }
+    case actionTypes.CREATE_NEW_STORY_CARD: {
+      const { stories } = state.draft;
+      const { file } = action.args;
+      return {
+        ...state,
+        draft: {
+          ...state.draft,
+          stories: [...stories, {
+            ...newStory(),
```
Comment (MayaUribe) [inline packages/story-group-composer/reducer.js]: Oh I see, this is really really great! we should try to implement something like this for other reducers ✨

File: packages/story-group-composer/index.js (hunk 3 above)
Comment (karelm) [reply packages/story-group-composer/index.js]: yup, thats the one 😄

File: packages/story-group-composer/reducer.js (hunk 4 above)
Comment (karelm) [reply packages/story-group-composer/reducer.js]: I'd still love to have immutable data everywhere in the ideal world, but for now this works 💯

**Pull request merged**